AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngineAdapter.swift

```diff
@@ -15,14 +15,17 @@ protocol StorageEngineAdapter: class, ModelStorageBehavior {
     func save(untypedModel: Model, completion: @escaping DataStoreCallback<Model>)
 
     func delete<M: Model>(_ modelType: M.Type,
+                          modelSchema: ModelSchema,
```

Comment:
I'm a little bit sad that we have to add the `modelSchema` parameters even though it's an internal-only protocol. Is there any benefit to making the old signatures default `modelSchema` to `modelType.schema` for these, to support use cases like we have below where we specify `MutationEvent.self` and `MutationEvent.schema`?

Reply:
Cannot give default value in the protocol, so it will be difficult to enforce it. Also if we add default value, there would be a chance that the ModelSchema is not passed in correctly. So I guess keeping it separate will be the safer approach for now.

Reply:
I'm not suggesting to add a default value, I'm suggesting two method signatures with two implementations, one that derives the default value from the incoming `model` and one that is specified. The one that derives would of course call the specific one.

Reply:
Updated as per our conversation, added default methods in the protocol.